src/unitycatalog/ai/utils.py

```diff
@@ -113,3 +141,148 @@ def validate_full_function_name(function_name: str) -> FullFunctionName:
             f"Invalid function name: {function_name}, expecting format <catalog_name>.<schema_name>.<function_name>."
         )
     return FullFunctionName(catalog_name=splits[0], schema_name=splits[1], function_name=splits[2])
+
+
+def uc_type_json_to_pydantic_type(uc_type_json: Union[str, Dict]) -> Type:
+    """
+    Convert Unity Catalog type json to Pydantic type.
+    """
+    if isinstance(uc_type_json, str):
+        type_name = uc_type_json.upper()
+        if type_name in UC_TYPE_JSON_MAPPING:
+            return Union[UC_TYPE_JSON_MAPPING[type_name]]
+        # the type text contains the precision and scale
+        elif type_name.startswith("DECIMAL"):
+            return Union[decimal.Decimal, float]
+        else:
+            raise TypeError(
+                f"Type {uc_type_json} is not supported. Supported "
+                f"types are: {UC_TYPE_JSON_MAPPING.keys()}"
+            )
+    if isinstance(uc_type_json, dict):
+        tpe = uc_type_json["type"]
+        if tpe == "array":
+            element_type = uc_type_json_to_pydantic_type(uc_type_json["elementType"])
+            if uc_type_json["containsNull"]:
+                element_type = Optional[element_type]
+            return Union[List[element_type], Tuple[element_type]]
+        elif tpe == "map":
+            key_type = uc_type_json["keyType"]
+            if key_type != "string":
+                raise TypeError(f"Only support STRING key type for MAP but got {key_type}.")
+            value_type = uc_type_json_to_pydantic_type(uc_type_json["valueType"])
+            if uc_type_json["valueContainsNull"]:
+                value_type = Optional[value_type]
+            return Dict[str, value_type]
+        elif tpe == "struct":
+            fields = {}
+            for field in uc_type_json["fields"]:
+                field_type = uc_type_json_to_pydantic_type(field["type"])
+                comment = field.get("metadata", {}).get("comment")
+                if field.get("nullable"):
+                    field_type = Optional[field_type]
+                    fields[field["name"]] = (field_type, Field(default=None, description=comment))
+                else:
+                    fields[field["name"]] = (field_type, Field(..., description=comment))
+            uc_type_json_str = json.dumps(uc_type_json, sort_keys=True)
+            type_hash = md5(uc_type_json_str.encode()).hexdigest()[:8]
```

Comment:
```suggestion
            type_hash = md5(uc_type_json_str.encode(), usedforsecurity=False).hexdigest()[:8]
```

ref: https://github.com/mlflow/mlflow/pull/9961/